src/mngt/ocf_mngt_cache.c

```diff
@@ -1347,6 +1347,18 @@ static void _ocf_mngt_attach_init_instance(ocf_pipeline_t pipeline,
 {
 	struct ocf_cache_attach_context *context = priv;
 	ocf_cache_t cache = context->cache;
+	int result;
+
+	if (!context->flags.cleaner_started) {
```

Comment:
Is it possible that cleaner is somehow started earlier? I see this is the only function where it's started, which means that it would have to be called twice during attach procedure, which is not the case. Hence I think this check is unnecessary.